web/server/codechecker_server/api/tasks.py

```diff
@@ -0,0 +1,391 @@
+            raise ValueError("Task self-failure as per the user's request.")
+
+
+def _db_timestamp_to_posix_epoch(d: Optional[datetime.datetime]) \
+        -> Optional[int]:
+    return int(d.replace(tzinfo=datetime.timezone.utc).timestamp()) if d \
+        else None
+
+
+def _posix_epoch_to_db_timestamp(s: Optional[int]) \
+        -> Optional[datetime.datetime]:
+    return datetime.datetime.fromtimestamp(s, datetime.timezone.utc) if s \
+        else None
+
+
+def _make_task_info(t: DBTask) -> TaskInfo:
+    """Format API `TaskInfo` from `DBTask`."""
+    return TaskInfo(
+        token=t.token,
+        taskKind=t.kind,
+        status=TaskStatus._NAMES_TO_VALUES[t.status.upper()],
+        productId=t.product_id or 0,
+        actorUsername=t.username,
+        summary=t.summary,
+        comments=t.comments,
+        enqueuedAtEpoch=_db_timestamp_to_posix_epoch(t.enqueued_at),
+        startedAtEpoch=_db_timestamp_to_posix_epoch(t.started_at),
+        completedAtEpoch=_db_timestamp_to_posix_epoch(t.finished_at),
+        lastHeartbeatEpoch=_db_timestamp_to_posix_epoch(
+            t.last_seen_at),
+        cancelFlagSet=t.cancel_flag,
+    )
+
+
+def _make_admin_task_info(t: DBTask) -> AdministratorTaskInfo:
+    """Format API `AdministratorTaskInfo` from `DBTask`."""
+    return AdministratorTaskInfo(
+        normalInfo=_make_task_info(t),
+        machineId=t.machine_id,
+        statusConsumed=t.consumed,
+    )
+
+
+# These names are inherited from Thrift stubs.
+# pylint: disable=invalid-name
+class ThriftTaskHandler:
+    """
+    Manages Thrift requests concerning the user-facing Background Tasks API.
+    """
+
+    def __init__(self,
+                 configuration_database_sessionmaker,
+                 task_manager: TaskManager,
+                 auth_session):
+        self._config_db = configuration_database_sessionmaker
+        self._task_manager = task_manager
+        self._auth_session = auth_session
+
+    def _get_username(self) -> Optional[str]:
+        """
+        Returns the actually logged in user name.
+        """
+        return self._auth_session.user if self._auth_session else None
+
+    @exc_to_thrift_reqfail
+    @timeit
+    def getTaskInfo(self, token: str) -> TaskInfo:
+        """
+        Returns the `TaskInfo` for the task identified by `token`.
+        """
+        with DBSession(self._config_db) as session:
+            db_task: Optional[DBTask] = session.query(DBTask).get(token)
+            if not db_task:
+                raise RequestFailed(ErrorCode.GENERAL,
+                                    f"Task '{token}' does not exist!")
+
+            has_right_to_query_status: bool = False
+            should_set_consumed_flag: bool = False
+
+            if db_task.username == self._get_username():
+                has_right_to_query_status = True
+                should_set_consumed_flag = db_task.is_in_terminated_state
+            elif db_task.product_id is not None:
```

Comment:
Because different workflows are executed. You can only query the status of the task if:

1. You are the user who was responsible for creating the task. (In this case, but only for terminated tasks, the `consumed` flag is set to true.)
2. The task is associated with a product (this might not always be the case!) and you are an admin of said product. (In this case, the `consumed` flag is left intact.)
3. You are a superuser. (In this case, the `consumed` flag is left intact.)

If this was not an `elif`, a product admin or superuser querying their own task would leave the `consumed` flag intact.